docker/autoware-universe/Dockerfile

```diff
@@ -58,14 +58,70 @@ RUN mkdir -p /etc/OpenCL/vendors \
 RUN echo "source /opt/ros/${ROS_DISTRO}/setup.bash" > /etc/bash.bashrc
 CMD ["/bin/bash"]
 
-FROM devel as prebuilt
+FROM devel as builder
 SHELL ["/bin/bash", "-o", "pipefail", "-c"]
 
 ## Build and change permission for runtime data conversion
 RUN source /opt/ros/"$ROS_DISTRO"/setup.bash \
   && colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release \
   && find /autoware/install -type d -exec chmod 777 {} \;
 
+ARG CUDA_BASE_IMAGE
```

Comment:
It seems this line should come at the beginning of the file.
Did it work on your environment?

```console
❯ ./docker/build.sh
+ docker buildx bake --no-cache --load --progress=plain -f /home/kenji/autoware/docker/autoware-universe/docker-bake.hcl --set '*.context=/home/kenji/autoware/docker/../' --set '*.ssh=default' --set '*.platform=linux/amd64' --set '*.args.ROS_DISTRO=humble' --set '*.args.BASE_IMAGE=ubuntu:22.04' --set '*.args.SETUP_ARGS=--no-cuda-drivers' --set '*.args.CUDA_BASE_IMAGE=ubuntu:22.04' --set devel.tags=ghcr.io/autowarefoundation/autoware-universe:humble-latest-cuda --set prebuilt.tags=ghcr.io/autowarefoundation/autoware-universe:humble-latest-prebuilt-cuda
#1 [devel internal] load build definition from Dockerfile
#1 transferring dockerfile: 4.34kB done
#1 DONE 0.0s

#2 [prebuilt internal] load build definition from Dockerfile
#2 transferring dockerfile: 4.34kB done
#2 DONE 0.0s

#3 [devel internal] load .dockerignore
#3 transferring context: 2B done
#3 DONE 0.0s

#4 [prebuilt internal] load .dockerignore
#4 transferring context: 2B done
#4 DONE 0.0s
ERROR: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to create LLB definition: base name ($CUDA_BASE_IMAGE) should not be blank
```

Reply:
Mirrored the branch in the AWF, and ran the CI workflow.
https://github.com/autowarefoundation/autoware/tree/2840-reduce-prebuilt-docker-image-size
https://github.com/autowarefoundation/autoware/actions/runs/4158993011

But it has failed.
https://github.com/autowarefoundation/autoware/actions/runs/4158993011/jobs/7194629355#step:5:739

Reply:
Hi, @kenji-miyake , I have pushed a new commit about this docker build issue, could you please help to update the mirrored branch in AWF(https://github.com/autowarefoundation/autoware/tree/2840-reduce-prebuilt-docker-image-size), so we can retry the CI workflow.

Reply:
Updated and re-ran the workflow.
https://github.com/autowarefoundation/autoware/actions/runs/4195074077

Reply:
IF you want an ARG take effect in all FROMs, you should define it before the first FROM, and it will take effect in FROMs only.
`--------------------
   1 |     ARG BASE_IMAGE
   2 |     # hadolint ignore=DL3006
   3 | >>> FROM $BASE_IMAGE as devel
   4 |     SHELL ["/bin/bash", "-o", "pipefail", "-c"]
   5 |     
--------------------
ERROR: failed to solve: base name ($BASE_IMAGE) should not be blank
Error: buildx bake failed with: ERROR: failed to solve: base name ($BASE_IMAGE) should not be blank`

Reply:
@caiyuexiang thanks for your information. 
Hi, @kenji-miyake , could you please help to update and re-run the workflow?  We have tried to build with the Dockerfile locally, but it always failed with timeout.

Reply:
@Sharrrrk Sure, I did it. But I guess you can trigger the workflow on your fork as well because it's not the self-hosted machine. Also, you probably have write access to this repository because you're one of the maintainers.
https://github.com/autowarefoundation/autoware/actions/runs/4209409696

Reply:
Hi, @kenji-miyake Thanks for your help. After sync the fork from AWF, the workflow shown up under the Action Menu. And now, it has played and successed on my fork, please help to review it  again, thank you.

Reply:
@Sharrrrk Thank you! I'm sorry to be late.

This one? I'd appreciate it if you could write the link. :bow: 
https://github.com/Sharrrrk/autoware/actions/runs/4239792898

I ran the workflow on this repository just in case.
https://github.com/autowarefoundation/autoware/actions/runs/4288464786